File: common/src/main/java/dev/obscuria/fragmentum/api/Fragmentum.java
package dev.obscuria.fragmentum.api;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.ServiceLoader;

public final class Fragmentum
{
    public static final String MODID = "fragmentum";
    public static final String DISPLAY_NAME = "Fragmentum";
    public static final Logger LOG = LoggerFactory.getLogger(DISPLAY_NAME);
    public static final IPlatform PLATFORM = ServiceLoader.load(IPlatform.class).findFirst().orElseThrow();
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/api/Fragmentum.kt
package dev.obscuria.fragmentum.api

import dev.obscuria.fragmentum.api.service.FragmentumServices
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.util.ServiceLoader

object Fragmentum {

    const val MODID: String = "fragmentum"
    const val DISPLAY_NAME: String = "Fragmentum"
    @JvmStatic
    val LOGGER: Logger = LoggerFactory.getLogger(DISPLAY_NAME)
    @JvmStatic
    val PLATFORM: Platform = ServiceLoader.load(Platform::class.java).first()

    internal val SERVICES: FragmentumServices = ServiceLoader.load(FragmentumServices::class.java).first()
}